public/app/features/dashboard-scene/settings/GeneralSettings.tsx

```diff
@@ -22,11 +32,160 @@ export class GeneralSettingsEditView
   static Component = ({ model }: SceneComponentProps<GeneralSettingsEditView>) => {
     const dashboard = getDashboardSceneFor(model);
     const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+    const { title, description, tags, meta, editable, timepicker, timezone, weekStart, liveNow } = dashboard.useState();
+
+    const onTitleChange = (value: string) => {
+      dashboard.setState({ title: value });
+    };
+
+    const onDescriptionChange = (value: string) => {
+      dashboard.setState({ description: value });
+    };
+
+    const onTagsChange = (value: string[]) => {
+      dashboard.setState({ tags: value });
+    };
+
+    const onFolderChange = (newUID: string, newTitle: string) => {
+      const newMeta = {
+        ...meta,
+        folderUid: newUID || meta.folderUid,
+        folderTitle: newTitle || meta.folderTitle,
+        hasUnsavedFolderChange: true,
+      };
+
+      dashboard.setState({ meta: newMeta });
+    };
+
+    const onEditableChange = (value: boolean) => {
+      dashboard.setState({ editable: value });
+    };
+
+    const onTimeZoneChange = (value: TimeZone) => {
+      dashboard.setState({ timezone: value });
+    };
+
+    const onWeekStartChange = (value: string) => {
+      dashboard.setState({ weekStart: value });
+    };
+
+    const onRefreshIntervalChange = (value: string[]) => {
+      dashboard.setState({
+        timepicker: {
+          ...timepicker,
+          refresh_intervals: value,
+        },
+      });
+    };
```

Comment:
So here what you want to do is actually modify the state of an object that abstracts this behavior - in this case its RefreshPicker that's part of the DashboardScene controls. This would apply to other aspects of a dashboard that are represented via scene objects, i.e. SceneTimePicker.

Reply:
🤔 Interesting. I was wondering if we're looking for setting the state of the dashboard object, or this is the "settings" of the dashboard, which in the end is the dashboard itself.

AFAIK, this `timeseries` property stores the default configuration of the timepicker, but not the current time picker itself. This is why, in this case, I didn't opt to go for $timeRange. 

So, if I understand correctly, you mean I'm missing to apply this new state to RefreshPicker, or, in this case, RefreshPicker should consume this dashboard state to react to changes.

on it!

Reply:
Done!